src/Config.UserSecrets/PathHelper.cs

```diff
@@ -40,8 +40,18 @@ public static string GetSecretsPathFromSecretsId(string userSecretsId)
                         badCharIndex));
             }
 
-            var root = Environment.GetEnvironmentVariable("APPDATA") ??         // On Windows it goes to %APPDATA%\Microsoft\UserSecrets\
-                        Environment.GetEnvironmentVariable("HOME");             // On Mac/Linux it goes to ~/.microsoft/usersecrets/
+            const string userSecretsDir = "DOTNET_USER_SECRETS_ROOT";
```

Comment:
Is this a new fallback being added in this PR?

Reply:
I figured we should start by making it a fallback. Since usersecrets is called when the web server begins by default now, I want to give users an option to avoid their server crashing on boot  We can discuss making it an override with the PMs.

Reply:
Yeah maybe the name should have fallback in it if you decide to keep it as a fallback was my only point

Reply:
Let's change to DOTNET_USER_SECRETS_FALLBACK. We don't really want to encourage users to customize the location of user secrets. It's really mean to be an internal implementation detail.